### PR TITLE
Migrated changes that apply to loading model and optimizer

### PR DESCRIPTION
**Pull request by josephattieh**
This allows to load model and optimizer from checkpoint.
This does not reload the data state.

### CODE REVIEW

File: mammoth/models/model_saver.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import os
-from collections import deque
+from glob import glob
+from collections import deque, defaultdict
```
Comment (TimotheeMickus) [inline mammoth/models/model_saver.py:3]: defaultdict is imported but not used..?

File: mammoth/model_builder.py [hunk 2]
```diff
@@ -332,6 +350,9 @@ def build_task_specific_model(
     if uses_adapters(model_opts):
         logger.info('Creating adapters...')
         create_all_adapters(nmt_model, model_opts, task_queue_manager)
+        if checkpoint:
+            # TODO: plug in properly
```
Comment (TimotheeMickus) [inline mammoth/model_builder.py:354]: this TODO probably worth creating another issue

**Review by TimotheeMickus — COMMENTED** [review]
LGTM, but 
- you do have some linting errors that still need fixing
- would like a confirmation that this has been tested
- could benefit from setting up an explicit unit test

**Comment by josephattieh** [comment]
> LGTM, but
> 
> * you do have some linting errors that still need fixing
> * I would like a confirmation that this has been tested
> * this could benefit from setting up an explicit unit test

The linting errors have been fixed and the feature has been tested!

**Pull request merged**